src/iris/bin/sender.py

```diff
@@ -1120,7 +1145,18 @@ def main():
         cache.refresh()
         cache.purge()
 
-        if is_master:
+        # If we're currently a master, ensure our master-greenlets are running
+        # and we're doing the master duties
+        if coordinator.am_i_master():
+            if not bool(gwatch_renewer_task):
+                if should_mock_gwatch_renewer:
```

Comment:
instead of checking on if we should mock or not every loop have the main code check and swap out the function  with the mocked function.